Bottom: ac93da382386d7ab0239b28cdc00d6e3ed4470ea
Top:    3759227a3e0717b06c5ad98d24b12a338ac460d4
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-09-21 17:16:09 -0700

librdmacm: Place container_of inside #ifdef

verbs.h defines container_of.  Only define it if not defined

Signed-off-by: Sean Hefty <sean.hefty@intel.com>


---

diff --git a/src/cma.c b/src/cma.c
index d168df6..91bf108 100755
--- a/src/cma.c
+++ b/src/cma.c
@@ -124,8 +124,10 @@ static pthread_mutex_t mut = PTHREAD_MUTEX_INITIALIZER;
 static int abi_ver = RDMA_USER_CM_MAX_ABI_VERSION;
 int af_ib_support;
 
+#ifndef container_of
 #define container_of(ptr, type, field) \
 	((type *) ((void *)ptr - offsetof(type, field)))
+#endif
 
 static void ucma_cleanup(void)
 {
